c7n/resources/mq.py

```diff
@@ -83,10 +85,54 @@ def get_dimensions(self, resource):
                  'Value': "{}-1".format(resource['BrokerName'])}]
 
 
+@MessageBroker.filter_registry.register('vpc')
+class VpcFilter(VpcFilter):
+    """Filter a resource by its VPC id. Choose to select resources that are in that VPC or select
+    resources that are not in that VPC using the 'op' field.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: mq-vpc-filters
+                resource: message-broker
+                filters:
+                      - type: vpc
+                        key: VpcId
+                        value: vpc-xxxxxxxxxxx
+                        op: eq
+    """
+
+    def get_related_ids(self, resources):
+        client = local_session(self.manager.session_factory).client('ec2')
+        related_ids = set()
+        sg_ids = set()
+        for r in resources:
+            sg_id = r.get('SecurityGroups', [])[0]
+            vpc_id = self.manager._cache.get(sg_id)
```

Comment:
don't use the cache directly, build up a list of the resource ids you want to fetch and then use self.get_resource_manager(other_resource_type).get_resources(resource_ids)

Reply:
Hey @kapilt, I've been looking at this for the past couple of days, but I'm a little confused by your suggested implementation. If I use self.get_resource_manager().get_resources(resource_ids), it just returns the full set of VPC objects, but what I was originally doing with the cache was saving a security group ID with it's corresponding VPC ID (since I'm using security group IDs to look up which VPC each broker is in). I don't see how I can use this new implementation since it isn't relying on security group IDs for lookups anymore.